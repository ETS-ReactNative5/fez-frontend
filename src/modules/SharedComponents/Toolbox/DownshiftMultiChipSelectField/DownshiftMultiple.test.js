import {
    DownshiftMultiple,
    styles
} from './DownshiftMultiple';

const setup = (testProps, isShallow = true) => {
    const props = {
        classes: {},
        optionsList: [],
        ...testProps
    }
    return getElement(DownshiftMultiple, props, isShallow);
};

describe('DownshiftMultiple component', () => {
    it('should render properly', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('Downshift').props().itemToString()).toBe('');
    });

    it('should handleKeyDown(event)', () => {
        const wrapper = setup({});
        wrapper.setState({
            inputValue: '',
            selectedItems: ['test1', 'test2']
        });
        wrapper.instance().handleKeyDown({ which: 8 });
        expect(wrapper.state().selectedItems).toEqual(['test1']);
        wrapper.instance().handleKeyDown({ which: 65 });
        expect(wrapper.state().selectedItems).toEqual(['test1']);
    });

    it('should handleInputChange(event)', () => {
        const testFn = jest.fn();
        const testArr = [{}];
        const wrapper = setup({
            onChange: testFn
        });
        const event = {
            target: {
                value: 'test'
            }
        };
        wrapper.setState({
            selectedItems: testArr
        });
        wrapper.instance().handleInputChange(event);
        expect(testFn).toBeCalledWith(testArr);
    });

    it('should handleChange()', () => {
        const testFn = jest.fn();
        const wrapper = setup({
            onChange: testFn
        });
        wrapper.setState({
            inputValue: 'something',
            selectedItems: ['test1']
        });
        wrapper.instance().handleChange('test2');
        expect(wrapper.state().selectedItems).toEqual(['test1', 'test2']);
        expect(wrapper.state().inputValue).toBe('');
        expect(testFn).toBeCalledWith(['test1', 'test2']);

        testFn.mockClear();
        wrapper.setState({
            inputValue: 'something'
        });
        wrapper.instance().handleChange('test2');
        expect(wrapper.state().selectedItems).toEqual(['test1', 'test2']);
        expect(wrapper.state().inputValue).toBe('');
        expect(testFn).toBeCalledWith(['test1', 'test2']);
    });

    it('should handleDelete()', () => {
        const testFn = jest.fn();
        const wrapper = setup({
            onChange: testFn
        });
        wrapper.setState({
            selectedItems: ['test1', 'test2']
        });
        wrapper.instance().handleDelete('test2')();
        expect(wrapper.state().selectedItems).toEqual(['test1']);
        expect(testFn).toBeCalledWith(['test1']);
    });

    it('should handleClearAll()', () => {
        const testFn = jest.fn();
        const wrapper = setup({
            onChange: testFn
        });
        wrapper.setState({
            selectedItems: ['test1', 'test2']
        });
        wrapper.instance().handleClearAll();
        expect(wrapper.state().selectedItems).toEqual([]);
        expect(testFn).toBeCalledWith([]);
    });

    it('should sendData()', () => {
        const testFn = jest.fn();
        const wrapper = setup({
            onChange: testFn
        });
        const test = {};
        wrapper.instance().sendData(test);
        expect(testFn).toBeCalledWith(test);

        testFn.mockClear();
        wrapper.instance().sendData();
        expect(testFn).not.toBeCalled();
    });
});

describe('styles helper', () => {
    it('should return styles object as expected', () => {
        const test = styles({
            spacing: {
                unit: 4
            }
        });
        expect(test).toMatchSnapshot();
    });
});

