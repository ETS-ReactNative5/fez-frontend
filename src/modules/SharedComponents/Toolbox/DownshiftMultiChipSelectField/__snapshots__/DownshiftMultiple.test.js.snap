// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownshiftMultiple component should render properly 1`] = `
<Downshift
  defaultHighlightedIndex={null}
  defaultInputValue=""
  defaultIsOpen={false}
  defaultSelectedItem={null}
  environment={[Window]}
  getA11yStatusMessage={[Function]}
  id="downshift-multiple"
  inputValue=""
  itemToString={[Function]}
  onChange={[Function]}
  onInputValueChange={[Function]}
  onOuterClick={[Function]}
  onSelect={[Function]}
  onStateChange={[Function]}
  onUserAction={[Function]}
  scrollIntoView={[Function]}
  selectedItemChanged={[Function]}
  selectedItems={Array []}
  stateReducer={[Function]}
  suppressRefError={false}
>
  <Component />
</Downshift>
`;

exports[`styles helper should return styles object as expected 1`] = `
Object {
  "chip": Object {
    "margin": "2px 1px",
  },
  "clearAll": Object {
    "&:hover": Object {
      "opacity": 1,
    },
    "marginBottom": -12,
    "marginRight": -12,
    "marginTop": 12,
    "opacity": 0.66,
  },
  "container": Object {
    "flexGrow": 1,
    "position": "relative",
  },
  "divider": Object {
    "height": 8,
  },
  "inputInput": Object {
    "flexGrow": 1,
    "overflow": "hidden",
    "textOverflow": "ellipsis",
    "width": "auto",
  },
  "inputRoot": Object {
    "flexWrap": "wrap",
  },
  "paper": Object {
    "left": 0,
    "marginTop": 4,
    "position": "absolute",
    "right": 0,
    "zIndex": 1,
  },
  "root": Object {
    "flexGrow": 1,
    "height": 250,
  },
}
`;
